react.js

```diff
@@ -61,6 +61,8 @@ module.exports = {
     'react/jsx-uses-vars': 'warn',
     // Prevent missing parentheses around multilines JSX
     'react/jsx-wrap-multilines': 'warn',
+    //  Prevent using Array index in key props
+    'react/no-array-index-key': 'warn',
```

Comment:
First recent use case that came to mind is if you have some constant array that you're using to generate lists of elements from in multiple places. Example being navigation through a wizard. You may have a sidebar, and some forward / back controls that need the same data:

```javascript
const steps = [
  { caption: 'First Page', path: 'wizard/first_page' },
  { caption: 'Second Page', path: 'wizard/second_page' }
]
```
And in 2 or more other places:
```javascript
<div>
  {steps.map((step, index) => <NavItem {...step} key={index} />)
</div>
```

The data doesn't change, so using the index is sufficient IMO. Also, sometimes the dataset doesn't naturally have a specific attribute that you could count on being unique, so it would be awkward to hard code something for no other reason when the data is constant. I feel like I run into this sort of thing quite often.